src/constraint_solver/io.cc

```diff
@@ -180,6 +188,16 @@ class FirstPassVisitor : public ModelVisitor {
   }
 
  private:
+  //// TODO: TBD: not sure we are confusing terms with IntExpr, losing IntVar-iness, nevermind Boolean, etc...
```

Comment:
We are not loosing intvar-ness, not boolean. This is taken care in the Solver::IntVar() method